src/main/java/com/yelp/nrtsearch/server/luceneserver/field/IndexableFieldDef.java

```diff
@@ -157,7 +173,13 @@ protected FacetValueType parseFacetValueType(Field requestField) {
    * @param fieldType type that needs search properties set
    * @param requestField field from request
    */
-  protected void setSearchProperties(FieldType fieldType, Field requestField) {}
+  protected void setSearchProperties(FieldType fieldType, Field requestField) {
+    if (this instanceof Keyable) {
+      fieldType.setIndexOptions(IndexOptions.DOCS);
+      fieldType.setTokenized(false);
+      fieldType.setOmitNorms(true);
+    }
+  }
```

Comment:
I'm not sure this is the best place to put this. The subclasses could change these values without realizing it. Also, this is only needed for the field instance that is actually the key. I would say move this logic into the subclass implementations.

Reply:
I have added a `setKeyableSearchProperties` to make sure that some they are set irrespective of subclass implementations. This is to make sure that the key doesn't have tokenization (like ATOM fields), and is also indexed .